### PR TITLE
Feature/4.3 mm book details

### PR DESCRIPTION
**Pull request by Kimara1904**
Implement Book Details
Implement Update and Delete from Book Details

### CODE REVIEW

File: fe-library-app/src/components/BookDetail/BookDetail.module.css [hunk 1]
```diff
@@ -0,0 +1,82 @@
+@import '../../variables.css';
+.book_details table{
+    width: 250px;
+    height: 250px;
+}
+
+.book_details img{
+    margin-top: 50px;
+    width: 250px;
+    height: 202px;
```
Comment (bushinSandra) [inline fe-library-app/src/components/BookDetail/BookDetail.module.css:10]: Why 202?

File: fe-library-app/src/components/BookDetail/BookDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import DefaultBookCover from '../BookList/BookCard/DefaultBookCover.png'
+import style from './BookDetail.module.css'
+
+const BookDetail = () => {
+  const { id } = useParams()
+  const [ book, setBook ] = useState<BookByIdItemResponse>()
+  const navigate = useNavigate()
+  const [ showUpdateBookModal, setShowUpdateBookModal ] = useState(false)
+
+  const getBookInfo = useCallback(() => {
```
Comment (bushinSandra) [inline fe-library-app/src/components/BookDetail/BookDetail.tsx:16]: You can pass id as function prop and not pass it in dependencies array, that way you do not recreate function on id change

File: fe-library-app/src/components/BookDetail/BookDetail.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        alert('Error with getting book details')
+      })
+  }, [ id ])
+
+  useEffect(() => {
+    getBookInfo()
+  }, [ getBookInfo ])
+
+  const handleModifyBookClick = () => {
+    navigate('/add_modify/' + (id as string))
```
Comment (bushinSandra) [inline fe-library-app/src/components/BookDetail/BookDetail.tsx:31]: Template string would be more readable here

File: fe-library-app/src/components/BookDetail/BookDetail.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  }
+
+  const handleBookDelete = () => {
+    if (confirm('Are you sure that you want to delete this book?') === true) {
+      deleteBook(id as string)
```
Comment (bushinSandra) [inline fe-library-app/src/components/BookDetail/BookDetail.tsx:36]: Prettier/better ways to convert number to string https://stackoverflow.com/questions/5765398/whats-the-best-way-to-convert-a-number-to-a-string-in-javascript

File: fe-library-app/src/components/BookDetail/BookDetail.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                  <td />
+                ) : (
+                  <td>
+                    {(book?.Authors[0].Firstname as string) +
+                      ' ' +
+                      (book?.Authors[0].Lastname as string)}
+                  </td>
+                )}
+              </tr>
+              {(book?.Authors.length as number) - 1 !== 0 && (
```
Comment (bushinSandra) [inline fe-library-app/src/components/BookDetail/BookDetail.tsx:100]: What if book.Authors.length is 0?

File: fe-library-app/src/components/BookDetail/BookDetail.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                  ))}
+                </>
+              )}
+            </tbody>
+          </table>
+          {showUpdateBookModal && (
+            <CreateUpdateBookModal
+              onCreateOrModifySuccess={() => getBookInfo()}
+              onHideModal={() => setShowUpdateBookModal(false)}
+              id={parseInt(id as string)}
```
Comment (bushinSandra) [inline fe-library-app/src/components/BookDetail/BookDetail.tsx:117]: Id is string or number or..? if it is number why you use parseInt and if it is string why do you set type as string?

**Review by bushinSandra — CHANGES_REQUESTED** [review]

File: fe-library-app/src/components/BookDetail/BookDetail.module.css (hunk 1 above)
Comment (Kimara1904) [reply fe-library-app/src/components/BookDetail/BookDetail.module.css:10]: It's pixel perfect

File: fe-library-app/src/components/BookDetail/BookDetail.tsx (hunk 2 above)
Comment (Kimara1904) [reply fe-library-app/src/components/BookDetail/BookDetail.tsx:16]: Done.

File: fe-library-app/src/components/BookDetail/BookDetail.tsx (hunk 3 above)
Comment (Kimara1904) [reply fe-library-app/src/components/BookDetail/BookDetail.tsx:31]: Done.

File: fe-library-app/src/components/BookDetail/BookDetail.tsx (hunk 5 above)
Comment (Kimara1904) [reply fe-library-app/src/components/BookDetail/BookDetail.tsx:100]: Fixed

**Pull request merged**